File: android/libapp/src/main/java/com/topdon/ble/callback/ScanListener.java
package com.topdon.ble.callback;

import android.Manifest;

import com.topdon.ble.Device;

public interface ScanListener {
    
    int ERROR_LACK_LOCATION_PERMISSION = 0;
    
    int ERROR_LOCATION_SERVICE_CLOSED = 1;
    
    int ERROR_SCAN_FAILED = 2;

    void onScanStart();

    void onScanStop();

    void onScanResult(Device device, boolean isConnectedBySys);

    void onScanError(int errorCode, String errorMsg);
}

File: android/libapp_backup_1756584390/src/main/java/com/topdon/ble/callback/ScanListener.kt
package com.topdon.ble.callback

import android.Manifest
import com.topdon.ble.Device

interface ScanListener {
    companion object {
        const val ERROR_LACK_LOCATION_PERMISSION = 0
        const val ERROR_LOCATION_SERVICE_CLOSED = 1
        const val ERROR_SCAN_FAILED = 2
    }

    fun onScanStart()
    fun onScanStop()
    fun onScanResult(device: Device, isConnectedBySys: Boolean)
    fun onScanError(errorCode: Int, errorMsg: String)
}
